### PR TITLE
#45 reworked to full junit 5 (removing mixed 4/5 tests)

### PR DESCRIPTION
**Pull request by FDelporte**

### CODE REVIEW

**Comment by FDelporte** [comment]
A lot of changes, but actually no code changes, only removing junit4 as there was a mixture of 4 and 5 in the project. 

Code changes = change of parameters of assert methods where the label needs to be at the last position instead of first.

File: pi4j-test/src/test/java/com/pi4j/test/platform/ManualPlatformsTest.java [hunk 1]
```diff
@@ -39,18 +42,18 @@
...
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
 
+@TestInstance(Lifecycle.PER_CLASS)
 public class ManualPlatformsTest {
 
     private Context pi4j;
 
-    @Before
+    @BeforeAll
```
Comment (eitch) [inline pi4j-test/src/test/java/com/pi4j/test/platform/ManualPlatformsTest.java:56]: This won't work, as @BeforeAll must be a static method:

https://junit.org/junit5/docs/current/user-guide/#writing-tests-annotations

File: pi4j-core/src/main/java/com/pi4j/runtime/impl/DefaultRuntime.java [hunk 2]
```diff
@@ -70,7 +67,6 @@
     private final RuntimeProperties properties;
     private final List<Plugin> plugins = new ArrayList<>();
     private boolean isShutdown = false;
-    private final ExecutorService executor = Executors.newSingleThreadExecutor();
```
Comment (eitch) [inline pi4j-core/src/main/java/com/pi4j/runtime/impl/DefaultRuntime.java:73]: The changes here seem to not be required for the JUnit changes, no? Should this be in a different pull request? Or why did you remove the shutting down?

**Review by eitch — CHANGES_REQUESTED** [review]
- The @BeforeAll must be changed, either to @BeforeEach, or the method must be static.
- The runtime changes shouldn't be in this pull request

File: pi4j-test/src/test/java/com/pi4j/test/platform/ManualPlatformsTest.java (hunk 1 above)
Comment (FDelporte) [reply pi4j-test/src/test/java/com/pi4j/test/platform/ManualPlatformsTest.java:56]: yes and no ;-)
by adding @TestInstance(Lifecycle.PER_CLASS) to the class, it is not needed to be static. 
static would also break the reference to private Context pi4j

File: pi4j-core/src/main/java/com/pi4j/runtime/impl/DefaultRuntime.java (hunk 2 above)
Comment (FDelporte) [reply pi4j-core/src/main/java/com/pi4j/runtime/impl/DefaultRuntime.java:73]: this slipped in while working (long time ago) on #18 and is still part of branch https://github.com/Pi4J/pi4j-v2/tree/feature/%2318. In that branch I also did the move to junit5 but extracted it into this separate branch so this could be merged independent of the work on the executorservice which actually still needs architectural investigation first.

File: pi4j-test/src/test/java/com/pi4j/test/platform/ManualPlatformsTest.java (hunk 1 above)
Comment (eitch) [reply pi4j-test/src/test/java/com/pi4j/test/platform/ManualPlatformsTest.java:56]: oh my... you are right. Sorry.

**Comment by eitch** [comment]
@FDelporte now there is conflict open. I guess after you rebase the feature branch with master then we can merge the pull request

**Comment by FDelporte** [comment]
@eitch conflict is fixed

**Review by savageautomate — APPROVED** [review]

**Review by eitch — APPROVED** [review]
